subsys/net/openthread/rpc/server/ot_rpc_meshdiag.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+#include "ot_rpc_resource.h"
+
+#include <nrf_rpc_cbor.h>
+
+#include <zephyr/net/openthread.h>
+
+#include <openthread/mesh_diag.h>
+
+/* Register iterators as resource table of size 1, to avoid passing raw pointers */
+OT_RPC_RESOURCE_TABLE_REGISTER(meshdiag_ip6_it, otMeshDiagIp6AddrIterator, 1);
+OT_RPC_RESOURCE_TABLE_REGISTER(meshdiag_child_it, otMeshDiagChildIterator, 1);
+
+static void ot_rpc_cmd_get_next_ip6_address(const struct nrf_rpc_group *group,
+					    struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	struct nrf_rpc_cbor_ctx rsp_ctx;
+	otError error;
+	otMeshDiagIp6AddrIterator *iterator;
+	ot_rpc_res_tab_key iterator_key;
+	otIp6Address address;
+
+	iterator_key = nrf_rpc_decode_uint(ctx);
+	iterator = ot_res_tab_meshdiag_ip6_it_get(iterator_key);
```

Comment:
`otMeshDiagGetNextIp6Address` doesn't accept NULL iterator (it only asserts this) so this should be checked at RPC layer similarly to decode failure.

Reply:
that's right. Had to make a small alignment to allow testing for that- moved msg res table declaration to `ot_rpc_resource.h` and added declaration for iterator tables there as well to make them accessible from test scope